tests/bluetooth/bsim_bt/bsim_test_l2cap/src/main_l2cap_ecred.c

```diff
@@ -0,0 +1,606 @@
+{
+	struct channel *chan = CONTAINER_OF(l2cap_chan, struct channel, le);
+
+	chan->buf = 0;
+	k_sem_give(&sent_sem);
+
+	LOG_DBG("chan_id: %d", chan->chan_id);
+}
+
+static void chan_connected_cb(struct bt_l2cap_chan *l2cap_chan)
+{
+	struct channel *chan = CONTAINER_OF(l2cap_chan, struct channel, le);
+
+	LOG_DBG("chan_id: %d", chan->chan_id);
+
+	LOG_DBG("tx.mtu %d, tx.mps: %d, rx.mtu: %d, rx.mps %d", sys_cpu_to_le16(chan->le.tx.mtu),
+		sys_cpu_to_le16(chan->le.tx.mps), sys_cpu_to_le16(chan->le.rx.mtu),
+		sys_cpu_to_le16(chan->le.rx.mps));
+
+	k_sem_give(&chan_conn_sem);
+
+	if (k_sem_count_get(&chan_conn_sem) == L2CAP_CHANNELS) {
+		k_sem_give(&all_chan_conn_sem);
+		k_sem_reset(&all_chan_disconn_sem);
+	}
+}
+
+static void chan_disconnected_cb(struct bt_l2cap_chan *l2cap_chan)
+{
+	struct channel *chan = CONTAINER_OF(l2cap_chan, struct channel, le);
+
+	LOG_DBG("chan_id: %d", chan->chan_id);
+
+	chan->in_use = false;
+	k_sem_take(&chan_conn_sem, K_FOREVER);
+
+	if (k_sem_count_get(&chan_conn_sem) == 0) {
+		k_sem_give(&all_chan_disconn_sem);
+		k_sem_reset(&all_chan_conn_sem);
+	}
+}
+
+static void chan_status_cb(struct bt_l2cap_chan *l2cap_chan, atomic_t *status)
+{
+	struct channel *chan = CONTAINER_OF(l2cap_chan, struct channel, le);
+
+	LOG_DBG("chan_id: %d, status: %ld", chan->chan_id, *status);
+}
+
+static void chan_released_cb(struct bt_l2cap_chan *l2cap_chan)
+{
+	struct channel *chan = CONTAINER_OF(l2cap_chan, struct channel, le);
+
+	LOG_DBG("chan_id: %d", chan->chan_id);
+}
+
+static void chan_reconfigured_cb(struct bt_l2cap_chan *l2cap_chan)
+{
+	struct channel *chan = CONTAINER_OF(l2cap_chan, struct channel, le);
+
+	LOG_DBG("chan_id: %d", chan->chan_id);
+}
+
+static const struct bt_l2cap_chan_ops l2cap_ops = {
+	.alloc_buf = chan_alloc_buf_cb,
+	.recv = chan_recv_cb,
+	.sent = chan_sent_cb,
+	.connected = chan_connected_cb,
+	.disconnected = chan_disconnected_cb,
+	.status = chan_status_cb,
+	.released = chan_released_cb,
+	.reconfigured = chan_reconfigured_cb,
+};
+
+static struct channel *get_free_channel(void)
+{
+	for (int idx = 0; idx < L2CAP_CHANNELS; idx++) {
+		struct channel *chan = &channels[idx];
+
+		if (chan->in_use) {
+			continue;
+		}
+
+		chan->chan_id = idx;
+		channels[idx].in_use = true;
+		(void)memset(chan->payload, idx, sizeof(chan->payload));
+		k_work_init(&chan->work, send_sdu_chan_worker);
+		chan->le.chan.ops = &l2cap_ops;
+		chan->le.rx.mtu = DATA_MTU;
+		chan->le.rx.mps = DATA_MPS;
+
+		return chan;
+	}
+
+	return NULL;
+}
+
+static void connect_num_channels(uint8_t num_l2cap_channels)
+{
+	struct bt_l2cap_chan *allocated_channels[ECRED_CHAN_MAX] = { NULL };
+
+	for (int i = 0; i < num_l2cap_channels; i++) {
+		struct channel *chan = get_free_channel();
+
+		if (!chan) {
+			FAIL("failed, chan not free");
+
+			return;
+		}
+
+		allocated_channels[i] = &chan->le.chan;
+	}
+
+	const int err = bt_l2cap_ecred_chan_connect(default_conn, allocated_channels,
+										servers[0].psm);
```

Comment:
The split line doesn't seem to be correctly aligned (should be N tabs + 0-7 spaces to align right after the opening parenthesis on the first line)